README.md

```diff
@@ -3,39 +3,77 @@
 A [Buildkite plugin](https://buildkite.com/docs/agent/v3/plugins) to allow pipeline steps to easily install
 private packages from an [npm](https://www.npmjs.com) repository.
 
-Note this plugin should work equally well despite your personal preferences for either `yarn` or `npm`.
+Note this plugin should work equally well despite any personal preferences for either `yarn` or `npm`.
 
 ## Example
 
-The following pipeline will run `yarn install` (and presumably some private packages).
+To read the value from an environment variable named `MY_TOKEN` when the plugin executes, use the `env:` prefix.
 
 ```yml
 steps:
   - command: yarn install
     plugins:
-      seek-oss/private-npm#v1.0.1:
-        token: ${NPM_TOKEN}
+      seek-oss/private-npm#v1.1.1:
+        token: "env:MY_TOKEN"
```

Comment:
How do you disambiguate tokens vs parameters for the token?

If you do this via YAML then you can have the best of all worlds:

`BUILDKITE_PLUGIN_PRIVATE_NPM_TOKEN`:
```
  seek-oss/private-npm#v1.1.1:
    token: "hard coded"
```

`BUILDKITE_PLUGIN_PRIVATE_NPM_TOKEN_ENV`:
```
  seek-oss/private-npm#v1.1.1:
    token:
      env: MY_TOKEN
```

`BUILDKITE_PLUGIN_PRIVATE_NPM_TOKEN_FILE`:
```
  seek-oss/private-npm#v1.1.1:
    token:
      file: "my-file"
```

Reply:
Yep that was my first instinct, but having three approaches means I can't make any one of them required in the `plugin.yml` file.  I think this is a better approach so perhaps it worth the trade off of having that extra layer of validation enforced by Buildkite

Reply:
I still think it's worth doing. None of the plugins seem to do "deep" schemas (e.g. what's in an `array` or that `string` can't be empty). The approach in this PR relies on inspecting the opaque token.

Reply:
IMO it's worth doing:

```
  properties:
    registry:
      type: string
    token:
      type: [string, object]
      properties:
        env:
          type: string
        file:
          type: string
``` 

and some basic validation in bash (e.g. one required). I really don't think the approach here is consistent with other Buildkite plugins.